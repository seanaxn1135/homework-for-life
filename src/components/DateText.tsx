import React from 'react';
import { Text, StyleSheet } from 'react-native';
import { colors } from '../theme/colors';

interface DateTextProps {
  style?: any;
}

const DateText: React.FC<DateTextProps> = ({ 
  style, 
}) => {
  const today = new Date();
  const formattedDateString = today.toLocaleDateString('en-US', {
    weekday: 'long',
    month: 'long',
    day: 'numeric',
  });

  return (
    <Text style={[styles.text, style]}>
      {formattedDateString}
    </Text>
  );
};

const styles = StyleSheet.create({
  text: {
    color: colors.textSubtle,
    fontSize: 14,
    fontFamily: 'Palanquin',
    lineHeight: 18,
    textAlign: 'center',
  },
});

export default DateText; 