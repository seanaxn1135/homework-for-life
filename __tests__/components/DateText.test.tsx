// __tests__/DateText.test.tsx

import React from 'react';
import { render, screen } from '@testing-library/react-native';
import DateText from '../../src/components/DateText';

const MOCK_CURRENT_DATE = new Date(2025, 4, 13); // May 13, 2025 (Tuesday)
const RealDate = Date;

beforeAll(() => {
  global.Date = class extends RealDate {
    constructor(...args: any[]) {
        super();
      if (args.length === 0) {
        return MOCK_CURRENT_DATE;
      }
      // @ts-ignore
      return new RealDate(...args);
    }

    static now() {
      return MOCK_CURRENT_DATE.getTime();
    }

    static parse(...args: Parameters<typeof RealDate.parse>) {
      return RealDate.parse(...args);
    }
    static UTC(...args: Parameters<typeof RealDate.UTC>) {
      return RealDate.UTC(...args);
    }
  } as any;
});

afterAll(() => {
  global.Date = RealDate;
});

describe('<DateText /> - Core Date Display', () => {
  it('should display the current (mocked) date formatted as "Weekday, Month Day"', () => {
    render(<DateText />);
    const expectedDateString = 'Tuesday, May 13';
    expect(screen.getByText(expectedDateString)).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { toJSON } = render(<DateText />);
    expect(toJSON()).toMatchSnapshot();
  });
});