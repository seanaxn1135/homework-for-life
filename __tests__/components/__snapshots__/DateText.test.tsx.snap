// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateText /> - Core Date Display should match snapshot 1`] = `
<Text
  style={
    [
      {
        "color": "#8A8A8A",
        "fontFamily": "Palanquin",
        "fontSize": 14,
        "lineHeight": 18,
        "textAlign": "center",
      },
      undefined,
    ]
  }
>
  Tuesday, May 13
</Text>
`;
